File: gcd/gcd.py
def gcd(a, b):
    controlValue = 0
    lower = 0
     
    if a > b:
        controlValue = a
        lower = b
    elif b > a:
        controlValue = b
        lower = a
    
    while True:
        controlValue = controlValue - lower

        if controlValue == 0: 
            gcd = lower
            return gcd

        if controlValue < lower:
            temp = lower
            lower = controlValue
            controlValue = temp

try:
    a = int(input("Enter first value: "))
    b = int(input("Enter second value: "))
except ValueError:
    print("Value is not a integer!")
    quit()

gdc(a, b)

        
    

File: gcd/gcd.cpp
#include <iostream>
#include <limits>
using namespace std;

int getNumberFromUser(){
    int a;
    while(!(cin >> a)){
        cout << "Value is not a integer!" << endl;
        cin.clear();
        cin.ignore(numeric_limits<streamsize>::max(), '\n');
    }

    return a;
}

int gcd(int a, int b) {
    int controlValue = 0, lower = 0, temp, nwd;

    if (a > b) {
        controlValue = a;
        lower = b;

    } else if(b > a) {
        controlValue = b;
        lower = a;

    }

    while (true) {
        controlValue = controlValue - lower;

        if (controlValue == 0) {
            nwd = lower;
            return nwd;

        }

        if (controlValue < lower) {
            temp = lower;
            lower = controlValue;
            controlValue = temp;

        }
    }
}

int main() {
    int a, b;

    cout << "Enter first number: ";
	a = getNumberFromUser();   

    cout << "Enter second value: ";
	b = getNumberFromUser();
        
    cout << "Greatest common divisor: " << gcd(a, b) << endl;
}

